lib/load.js

```diff
@@ -48,9 +48,17 @@ function load (source, options, defVal) {
     : null
 
   var opts = options || {}
-  return loader ? loader(source, opts)
+  var promise = loader ? loader(source, opts)
     : defVal ? Promise.resolve(defVal)
     : Promise.reject('Source not valid (' + source + ')')
+
+  return promise.then(function (data) {
+    opts.ready(null, data)
```

Comment:
I think there's a problem here: if options is not defined, then `opts = {}` and `opts.ready` will fail. I would prefer something like: 

```js
if (!opts.ready) return promise
else return promise.then(...)
````

or more idiomatic with the rest of the code:

```js
return opts.ready? promise.then(...) : promise
````

What do you think?